### PR TITLE
A4A: Register initial sections and routes.

### PR DESCRIPTION
**Pull request by jkguidaven**
This pull request includes the initial sections for the A4A portal. Please note. that this is not the final version and there may be additions in the future.

- Overview `/overview`
- Sites
  - Needs attentions `/sites?issue_types=all_issues`
  - Favorites `/sites/favorite`
  - All `/sites`
- Plugins `/plugins`

Closes https://github.com/Automattic/jetpack-genesis/issues/223

## Proposed Changes

* Add initial sections in a8c-for-agencies folder.
* Update `client/sections.js` to include new sections and routes
* Update the config to include new sections in the A4A dev environment only.

## Testing Instructions

* Checkout this branch
* Spin up the environment using the command yarn `start-a8c-for-agencies`.
* Test all the links above if it works.

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

### CODE REVIEW

File: client/a8c-for-agencies/sections/overview/index.tsx [hunk 1]
```diff
@@ -0,0 +1,7 @@
+import page from '@automattic/calypso-router';
+import { makeLayout, render as clientRender } from 'calypso/controller';
+import { overviewContext } from './controller';
+
+export default function () {
+	page( '/overview', overviewContext, makeLayout, clientRender );
```
Comment (yashwin) [inline client/a8c-for-agencies/sections/overview/index.tsx:6]: Can we check if the config is enabled first? This page will be accessible if the section is not enabled. 

```suggestion
    if ( config.isEnabled( 'section-name' ) ) {
	    page( '/overview', overviewContext, makeLayout, clientRender );
	 }
```

File: config/a8c-for-agencies-development.json [hunk 2]
```diff
@@ -19,7 +19,16 @@
...
+		"a8c-for-agencies-auth": true,
+		"a8c-for-agencies-marketplace": true,
+		"a8c-for-agencies-overview": true,
+		"a8c-for-agencies-plugins": true,
+		"a8c-for-agencies-purchases": true,
+		"a8c-for-agencies-referrals": true,
+		"a8c-for-agencies-resources": true,
+		"a8c-for-agencies-settings": true,
+		"a8c-for-agencies-sites": true,
+		"a8c-for-agencies-users": true
```
Comment (yashwin) [inline config/a8c-for-agencies-development.json]: I don't think we have to append every section with `a8c-for-agencies-`. We could probably do `navigation-item-` to be specific. What do you think?

**Review by yashwin — COMMENTED** [review]
Thanks for working on this, @jkguidaven!

I have a few things:

- Can we add the routes we are immediately planning to work on? For example we can only add overview, sites, etc. I feel like we will miss removing it if we plan to skip any page for MVP. Let's keep it minimal. 

I have also left a few comments. Let me know your thoughts.

File: client/a8c-for-agencies/sections/overview/index.tsx (hunk 1 above)
Comment (jkguidaven) [reply client/a8c-for-agencies/sections/overview/index.tsx:6]: I'm not sure if it's necessary. Shouldn't just including the section name in the environment config be enough? Right now, only the a4a environment config includes these sections, so they should only be accessible in that environment? no?

File: config/a8c-for-agencies-development.json (hunk 2 above)
Comment (jkguidaven) [reply config/a8c-for-agencies-development.json]: I believe it is necessary to append  `a8c-for-agencies-` in the section name to uniquely identify sections related to the A4A portal. Or I may have misunderstood your question.

Comment (yashwin) [reply config/a8c-for-agencies-development.json]: I mean everything here is related to the A4A portal, right? This is a nitpick; you can ignore it if you would like.

File: client/a8c-for-agencies/sections/overview/index.tsx (hunk 1 above)
Comment (yashwin) [reply client/a8c-for-agencies/sections/overview/index.tsx:6]: Yes, you are right. Sorry for that.

**Comment by jkguidaven** [comment]
> Can we add the routes we are immediately planning to work on? For example we can only add overview, sites, etc. I feel like we will miss removing it if we plan to skip any page for MVP. Let's keep it minimal.

My initial intention was to include as much boilerplate code now so we have something in place to work on later and also have a working demo of the navigation menu, but I do understand your point. For now, I'll leave the `overview`, `sites`, and `plugins`.

**Review by yashwin — APPROVED** [review]
Thanks for addressing my comments. Everything looks good now. 🚀

**Review by yashwin — APPROVED** [review]
Changes look good to me 🚀 

Could we please update the description according to the changes?

**Pull request merged**